src/storage/src/sst/parquet.rs

```diff
@@ -277,26 +269,27 @@ impl ParquetReader {
 
         let pruned_row_groups = self
             .predicate
-            .prune_row_groups(
-                store_schema.schema().clone(),
-                builder.metadata().row_groups(),
-            )
+            .prune_row_groups(builder.metadata().row_groups())
             .into_iter()
             .enumerate()
             .filter_map(|(idx, valid)| if valid { Some(idx) } else { None })
             .collect::<Vec<_>>();
 
         let parquet_schema_desc = builder.metadata().file_metadata().schema_descr_ptr();
 
-        let projection = ProjectionMask::roots(&parquet_schema_desc, adapter.fields_to_read());
+        let projection_mask = ProjectionMask::roots(&parquet_schema_desc, adapter.fields_to_read());
         let mut builder = builder
-            .with_projection(projection)
+            .with_projection(projection_mask.clone())
             .with_row_groups(pruned_row_groups);
 
         // if time range row filter is present, we can push down the filter to reduce rows to scan.
-        if let Some(row_filter) =
-            build_time_range_row_filter(self.time_range, &store_schema, &parquet_schema_desc)
-        {
+        if let Some(row_filter) = build_row_filter(
```

Comment:
question: Is the comment above still correct? Seems this function might return filters for other columns.